rom/laghos.cpp

```diff
@@ -1315,130 +1345,209 @@ int main(int argc, char *argv[])
+                SetWindowParameters(twparam, romOptions);
+            }
+
+            basis[0] = new ROM_Basis(romOptions, MPI_COMM_WORLD, sFactorX, sFactorV);
+            basis[0]->Init(romOptions, *S);
+
+            if (romOptions.mergeXV)
             {
-                // get out of the loop when no more file is found
-                last_step = true;
-                break;
+                romOptions.dimX = basis[0]->GetDimX();
+                romOptions.dimV = basis[0]->GetDimV();
+            }
+
+            int romSsize = romOptions.dimX + romOptions.dimV + romOptions.dimE;
+            romS.SetSize(romSsize);
+            if (infile_tw_steps.good())
+            {
+                infile_tw_steps >> nb_step;
             }
-            if (ti == nb_step) {
-                if (infile_tw_steps.good())
+            for (ti = 1; !last_step; ti++)
+            {
+                // romS = readCurrentReduceSol(ti);
+                // read ROM solution from a file.
+                // TODO: it needs to be read from the format of HDF5 format
+                // TODO: how about parallel version? introduce rank in filename
+                std::string filename = testing_parameter_outputPath + "/ROMsol/romS_" + std::to_string(ti);
+                std::ifstream infile_romS(filename.c_str());
+                if (infile_romS.good())
                 {
-                    infile_tw_steps >> nb_step;
-                }
-                romOptions.window++;
-                SetWindowParameters(twparam, romOptions);
-                basis[romOptions.window-1]->LiftROMtoFOM(romS, *S);
-                delete basis[romOptions.window-1];
-                basis[romOptions.window] = new ROM_Basis(romOptions, MPI_COMM_WORLD, sFactorX, sFactorV);
-                basis[romOptions.window]->Init(romOptions, *S);
+                    if ( (ti % vis_steps) == 0 )
+                    {
+                        if (myid == 0)
+                            cout << "Restoring " << ti << "-th solution" << endl;
+                        for (int k=0; k<romSsize; ++k)
+                        {
+                            infile_romS >> romS(k);
+                        }
+
+                        infile_romS.close();
+                        basis[romOptions.window]->LiftROMtoFOM(romS, *S);
 
-                if (romOptions.mergeXV)
+                        if (visit)
+                        {
+                            oper->ComputeDensity(*rho_gf);
+                            visit_dc->SetCycle(ti);
+                            visit_dc->SetTime(t);
+                            visit_dc->Save();
+                        }
+                    }
+                }
+                else
                 {
-                    romOptions.dimX = basis[romOptions.window]->GetDimX();
-                    romOptions.dimV = basis[romOptions.window]->GetDimV();
+                    // get out of the loop when no more file is found
```

Comment:
Exit the loop when no more files are found.

Reply:
Corrected.